demisto_sdk/commands/content_graph/common.py

```diff
@@ -281,6 +281,39 @@ class PackTags:
     DATA_SOURCE = "Data Source"
 
 
+class LazyProperty(property):
+    """
+    Used to define the properties which are lazy properties
+    """
+
+    pass
+
+
+def lazy_property(property_func: Callable):
+    """
+    lazy property: specifies that this property should be added to the pydantic model lazily
+    only when the instance property is first accessed.
+
+    Note:
+        make sure that the lazy property returns only serializable objects.
+
+    Use this decorator on your property in case you need it to be added to the model only if its called directly
+    """
+
+    def _lazy_decorator(self):
```

Comment:
what is the meaning of "self" outside a class scope?

Reply:
this decorator decorates methods of the `BaseContent` class (properties), hence the self here refers to the `self` of the decorator method from the `BaseContent`